### PR TITLE
Allow multiple conditions for expired quotes collection

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Description (*)

This is a refactoring of magento/magento2#20777

With Magento\Sales\Cron\CleanExpiredQuotes::setExpireQuotesAdditionalFilterFields() assigning values directly to the attribute and getExpireQuotesAdditionalFilterFields() being protected isn't callable outside the class, hence unable to add more to the existing condition, rather they end up being replaced with the new ones.

### Fixed Issues (if relevant)

-

### Manual testing scenarios (*)

-

### Questions or comments

EE code should also be adapted.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Foxtrot** [comment]
@magento run all test

**Comment by sivaschenko** [comment]
@p-bystritsky can you please sign the CLA

File: app/code/Magento/Sales/Model/ExpireQuotesFilterFieldsProvider.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Sales\Model;
+
+/**
+ * Provides expire quotes filter fields.
+ */
+class ExpireQuotesFilterFieldsProvider
```
Comment (swnsma) [inline app/code/Magento/Sales/Model/ExpireQuotesFilterFieldsProvider.php:13]: Do you really believe that we need to have a separated class to collect the filters?
Or could we just add filters as array parameter to the  `app/code/Magento/Sales/Model/ResourceModel/Collection/ExpiredQuotesCollection.php` ?
I can not come with real case when we need a new class.

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @p-bystritsky,
Thank you for you review!
Lets discuss your solution under my comment in the code.

**Comment by p-bystritsky** [comment]
Closing this PR due to creation of magento/magento2#24858.

**Pull request closed without merging**